### DISCUSSION TITLE
Can we document minimum typescript version support?

### DISCUSSION THREAD

**Discussion by equal-matt** [statelyai/xstate#3828] [Ideas]
Today I spent quite a while wrestling with the typescript compiler as a result of something in xstate. In the end, the problem was that I was on a version of Typescript that isn't compatible with xstate. Nowhere can I find anything on about Typescript version requirements, except for the [Typescript Changes](https://github.com/statelyai/xstate#typescript-changes) section of the README.

I'm not running a particularly old version of Typescript (4.6.2 - Feb '22) so I perhaps didn't reach for this as an option as quickly as I might have otherwise. It would have saved me a load of time if this was specified up front.

**Comment by equal-matt** [comment]
If anyone's interested in the specific issue - here's [a reproduction](https://stackblitz.com/edit/nextjs-7xzh9o?file=package.json) where you can run `npm run build` to experience the problem.

**Reply by Andarist** [reply]
Oh, I see the problem - I actually noticed it last week but didn't yet have time to fix it. Gonna put up a PR with a fix today.

**Comment by Andarist** [comment]
> Today I spent quite a while wrestling with the typescript compiler as a result of something in xstate.

Was this after upgrading to the very recent XState version (the one published today) or was it unrelated to that?

> I'm not running a particularly old version of Typescript (4.6.2 - Feb '22) so I perhaps didn't reach for this as an option as quickly as I might have otherwise. It would have saved me a load of time if this was specified up front.

This is a good point. I plan to set up CI pipeline that will test our types against a certain range of TS versions.

**Reply by equal-matt** [reply]
It was introducing xstate fresh to an existing project. I did this last week so not really related to today's version specifically. I did go through git blame, looking for when the specific change happened, at one point but decided it was a red herring.

**Reply by equal-matt** [reply]
I'm pro the CI pipeline ✅ 
Double points for using this to influence the install docs.

**Reply by Andarist** [reply]
> Double points for using this to influence the install docs.

Could you elaborate on this one? 🤔 I'm not sure if I understand what do you mean here.

**Reply by equal-matt** [reply]
The CI pipeline is good from the maintainers perspectives to stop regressions. I was just pointing out that your list of supported versions should _also_ go in the docs installation section, which is the bit I'm interested in as a consumer of the library. Not rocket science, just has the potential to be missed.
